test/self_protocore/propose_link.js

```diff
@@ -0,0 +1,158 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const BN = require('bn.js');
+
+const { AccountProvider } = require('../test_lib/utils.js');
+const web3 = require('../test_lib/web3.js');
+const Utils = require('../test_lib/utils.js');
+const ProtocoreUtils = require('../protocore/utils.js');
+const SelfProtocoreUtils = require('./utils.js');
+
+const TestSelfProtocore = artifacts.require('TestSelfProtocore');
+
+contract('SelfProtocore::proposeLink', (accounts) => {
+  const config = {};
+  const accountProvider = new AccountProvider(accounts);
+
+  beforeEach(async () => {
+    config.coconsensusAddress = accountProvider.get();
```

Comment:
Please move the deployment and setup code into util file.

Reply:
Done